lib/openstax/exchange/fake_client/fake_client.rb

```diff
@@ -43,20 +43,18 @@ def token
       end
 
       def create_identifier
-        SecureRandom.hex(64)
+        {
+          'read' => SecureRandom.hex(64),
+          'write' => SecureRandom.hex(64)
+        }
       end
 
       def record_multiple_choice_answer(identifier, resource, trial, answer)
-        @multiple_choice_responses[identifier] ||= {}
-        @multiple_choice_responses[identifier][resource] ||= {}
-
-        raise "invalid resource" \
-          unless URI(resource).host == "exercises.openstax.org"
-
-        raise "duplicate response for (identifier,resource,trial) triplet" \
-          if @multiple_choice_responses[identifier][resource][trial]
```

Comment:
Sending a new answer event to exchange with the same [identifier, resource, trial] should just append that event to the exercise activity for that [identifier, resource, trial].

Tutor controls when they can change the answer and should not send the answer to Exchange if the change is not allowed, but Exchange is unaware of that logic.